app/usedgoodreads/__main__.py

```diff
@@ -15,32 +15,59 @@ def index(key):
                         "price": v.price, "link": v.link} for v in books]
 
             results.sort(key=lambda each: each["price"])
 
-            return render_template("index.html", isbn=cache.isbn, title=cache.title,
+            return render_template("index.html", isbn=cache.isbn,
+                                   title=cache.title,
                                    results=results)
-
-    ticket = Ticket(key)
-
-    if q.fetch_job(ticket.isbn_jid):
-        return render_template("index.html")
-
-    if q.fetch_job(ticket.used_books_jid):
-        return render_template("index.html", status="searching")  # TODO: Should return actual status
-
-    q.enqueue(resolve_goodreads_key, ticket, job_timeout=60 * 1,
-            ttl=60 * 60, job_id=ticket.isbn_jid)
-
-    q.enqueue(resolve_used_books, key, job_timeout=60 * 1,
-              ttl=60 * 60, job_id=ticket.used_books_jid, depends_on=ticket.isbn_jid)
-
-    return render_template("index.html")
+        else:
+            # Used books need to be fetched
+            ticket = Ticket(key)
+            job = q.fetch_job(ticket.used_books_jid)
+            if job:
+                # Fetch job already in q
+                if job.get_status() == "finished":
+                    # There have been no results at the end of the fetch job
+                    return render_template("index.html",
+                                           isbn=cache.isbn,
+                                           title=cache.title,
+                                           status="Used books: No results")
+                else:
+                    # fetch job still running or there was an error
+                    return render_template("index.html",
+                                           isbn=cache.isbn,
+                                           title=cache.title,
+                                           status=f"Used books: {job.get_status()}")
+            else:
+                # Add fetch job to q
+                q.enqueue(resolve_used_books, key, job_timeout=60 * 1,
+                          ttl=60 * 60, job_id=ticket.used_books_jid)
+                return render_template("index.html",
+                                       isbn=cache.isbn,
+                                       title=cache.title,
+                                       status="Used books: enqueued")
+    else:
+        # Book is not in database
+        ticket = Ticket(key)
+        job = q.fetch_job(ticket.isbn_jid)
+        if job:
+            # Resolve task is already in q
+            return render_template("index.html",
+                                   status=f"URL: {job.get_status()}")
+        else:
+            # Add resolve task to q
+            q.enqueue(resolve_goodreads_key, ticket, job_timeout=60 * 1,
+                      ttl=60 * 60, job_id=ticket.isbn_jid)
+            return render_template("index.html", status="URL: enqueued")
+
+    abort(500)  # This should never happen
```

Comment:
Do you want to log this, too? So we have it in our logs and can grep for it.

Something like this should be good enough

    print("..", file=sys.stderr)

for logging right now.

Reply:
Fixed in new commit